expo/expo
# fix(metro-config): add `_expoRelativeProjectRoot` to invalidate Metro cache within monorepos

# Why

Fixes #29700 

# How

- Added `_expoRelativeProjectRoot` which is the project root, relative to the server root
- This is added as cache key to the Metro transformer, resulting in cache invalidation across workspace projects (without using absolute paths / is compatible cross-machine)

# Test Plan

- Clone repo from #29700 
- Start **apps/company** and open on iOS (without `--clear`)
- Close the server
- Start **apps/customer** and open on iOS (without `--clear`)
- Ensure the projects are not leaking into each other with Atlas
  - See https://github.com/expo/expo/issues/29700#issuecomment-2165348259 how to do that

# Checklist
